soc/wch/ch32v/common/qingke_v4/pinctrl_soc.h

```diff
@@ -10,7 +10,7 @@
 /**
  * @brief Type to hold a pin's pinctrl configuration.
  */
-struct ch32v208_pinctrl_soc_pin {
+struct qingke_v4_pinctrl_soc_pin {
```

Comment:
I don't think we should do this. The pinmux is a feature of the SOC, not the microarchitecture.

For example, both the CH32V003 and CH32V006 are QingKe V2, but they have different pinmux.

I'm concerned this PR will have the same issues as the previous one where there's lots of rebasing due to differing opinions. Should we work this out in a bug or chat first?